tests/test_stac_io.py

```diff
@@ -98,3 +98,19 @@ class ReportingStacIO(DefaultStacIO, DuplicateKeyReportingMixin):
                 str(excinfo.exception),
                 f'Found duplicate object name "key" in {src_href}',
             )
+
+    @unittest.mock.patch("pystac.stac_io.urlopen")
+    def test_headers_stac_io(self, urlopen_mock: unittest.mock.MagicMock) -> None:
+        stac_io = DefaultStacIO(headers={"Authorization": "api-key fake-api-key-value"})
+
+        try:
+            # note we don't care if this raises an exception, we just want to make
+            # sure urlopen was called with the appropriate headers
+            pystac.Catalog.from_file(
+                "https://example.com/catalog.json", stac_io=stac_io
+            )
+        except Exception:
+            pass
+
+        request_obj = urlopen_mock.call_args[0][0]
+        self.assertEqual(request_obj.headers, stac_io.headers)
```

Comment:
I played around with the mocks, and I think we can get rid of the blanket try/except by setting the `read` return value. Can you update your PR to look something like this:

```diff
diff --git a/tests/test_stac_io.py b/tests/test_stac_io.py
index 8609f4b..5314a96 100644
--- a/tests/test_stac_io.py
+++ b/tests/test_stac_io.py
@@ -1,6 +1,8 @@
+import json
 import os
-import unittest
 import tempfile
+import unittest
+import unittest.mock
 
 import pystac
 from pystac.stac_io import StacIO, DefaultStacIO, DuplicateKeyReportingMixin
@@ -103,14 +105,13 @@ class StacIOTest(unittest.TestCase):
     def test_headers_stac_io(self, urlopen_mock: unittest.mock.MagicMock) -> None:
         stac_io = DefaultStacIO(headers={"Authorization": "api-key fake-api-key-value"})
 
-        try:
-            # note we don't care if this raises an exception, we just want to make
-            # sure urlopen was called with the appropriate headers
-            pystac.Catalog.from_file(
-                "https://example.com/catalog.json", stac_io=stac_io
-            )
-        except Exception:
-            pass
+        catalog = pystac.Catalog("an-id", "a description").to_dict()
+        # required until https://github.com/stac-utils/pystac/pull/896 is merged
+        catalog["links"] = []
+        urlopen_mock.return_value.__enter__.return_value.read.return_value = json.dumps(
+            catalog
+        ).encode("utf-8")
+        pystac.Catalog.from_file("https://example.com/catalog.json", stac_io=stac_io)
 
         request_obj = urlopen_mock.call_args[0][0]
         self.assertEqual(request_obj.headers, stac_io.headers)
```

Reply:
Thanks for the suggestion, updated!